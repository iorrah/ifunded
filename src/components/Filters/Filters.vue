<template>
  <div class="filters container">
    <p class="filters__title">
      <img src="../../assets/filter-icon.svg" class="filters__icon" />
      Filter by
    </p>

    <div class="filters__inputs">
      <div class="material-form">
        <div class="material-form__input-text filters__input-text">
          <div class="group">
            <input
              data-message="The format of your email is incorrect"
              type="text"
              placeholder=""
              required=""
              class="inpField"
              :value="search"
              @input="updateSearch"
            />

            <img
              src="../../assets/search-icon.svg"
              class="material-form__search-icon"
            />

            <span class="highlight"></span>
            <span class="bar"></span>

            <label>
              Name or email
            </label>

            <span class="error-message">
              Error
            </span>
          </div>
        </div>

        <div class="material-form__input-select filters__input-select">
          <div class="group select-group">
            <div class="jq-selectbox jqselect inpField dropup opened">
              <select
                required="required"
                class="inpField inpField__select"
                @change="updateCreator($event)"
              >
                <option label="Everyone" value="0" selected="selected"></option>

                <option
                  v-for="creator in creators"
                  v-bind:key="creator.id"
                  v-bind:value="creator.id"
                  v-once
                >
                  {{creator.first_name}} {{creator.last_name}}
                </option>
              </select>

              <div class="jq-selectbox__select">
                <div class="jq-selectbox__select-text placeholder">
                  <span v-if="showCreatedBy">Created by</span>
                  <span v-else>
                    {{ creatorName }}
                  </span>
                </div>

                <div class="jq-selectbox__trigger">
                  <div class="jq-selectbox__trigger-arrow">
                    <img
                      src="../../assets/select-arrows.svg"
                      class="jq-selectbox__select-arrow"
                    />
                  </div>
                </div>
              </div>
            </div>

            <span class="highlight"></span>
            <span class="bar"></span>
          </div>
        </div>

        <div class="material-form__input-select filters__input-select">
          <div class="group select-group">
            <div class="jq-selectbox jqselect inpField dropup opened">
              <select
                required="required"
                class="inpField inpField__select"
                @change="updateMonth($event)"
              >
                <option value="0" selected="selected">Anytime</option>

                <option
                  v-for="month in months"
                  v-bind:key="month.id"
                  v-bind:value="month.id"
                  v-once
                >{{ month.name }} 2019</option>
              </select>

              <div class="jq-selectbox__select">
                <div class="jq-selectbox__select-text placeholder">
                  <span v-if="showCreationDate">Creation date</span>
                  <span v-else>
                    {{ getMonthById }} 2019
                  </span>
                </div>

                <div class="jq-selectbox__trigger">
                  <div class="jq-selectbox__trigger-arrow">
                    <img
                      src="../../assets/select-arrows.svg"
                      class="jq-selectbox__select-arrow"
                    />
                  </div>
                </div>
              </div>
            </div>

            <span class="highlight"></span>
            <span class="bar"></span>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import { months } from '../../utils/date-formatter';
import { isInvalidNumber } from '../../utils/validation';

export default {
  name: 'Filters',
  props: ['search', 'appUsers', 'creatorId', 'month'],
  computed: {
    creators() {
      const creatorsIds = this.appUsers.map((e) => e.creator_id);

      // eslint-disable-next-line arrow-body-style
      const uniqueIds = creatorsIds.filter((item, pos) => {
        return creatorsIds.indexOf(item) === pos;
      });

      // eslint-disable-next-line arrow-body-style
      const usersWhoCreated = uniqueIds.map((id) => {
        return this.appUsers.find((e) => e.id === id);
      });

      return usersWhoCreated || [];
    },
    showCreatedBy() {
      return this.$props.creatorId === null;
    },
    showCreationDate() {
      return isInvalidNumber(this.$props.month) || this.$props.month === 0;
    },
    creatorName() {
      const creator = this.creators.find((e) => e.id === this.$props.creatorId);

      if (!creator) {
        return '';
      }

      return `${creator.first_name} ${creator.last_name}`;
    },
    months() {
      return months;
    },
    getMonthById() {
      return months.find((e) => e.id === this.month).name;
    },
  },
  methods: {
    updateSearch(e) {
      this.$emit('update:search', e.target.value);
    },
    updateCreator(e) {
      const creatorId = e.target.value;

      if (creatorId === '0') {
        this.$emit('update:creatorId', null);
      } else {
        this.$emit('update:creatorId', parseInt(creatorId, 10));
      }
    },
    updateMonth(e) {
      const month = e.target.value;

      if (month === '0') {
        this.$emit('update:month', null);
      } else {
        this.$emit('update:month', parseInt(month, 10));
      }
    },
  },
};
</script>
