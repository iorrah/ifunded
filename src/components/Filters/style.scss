.filters {
  padding: 0.5rem 1.7rem;
  background: $light-gray-3;
  margin-bottom: 3.2rem;
}

.filters__title {
  font-size: 1rem;
  display: inline-block;
  margin: 0 3.7rem 0 0;
  color: $black;
}

.filters__icon {
  margin: 0 0.4rem 0 0;
}

.filters__inputs {
  display: inline-block;
  position: relative;
  top: -0.1rem;
}

.filters__input-text,
.filters__input-select {
  margin: 0 4rem 0 0;
}