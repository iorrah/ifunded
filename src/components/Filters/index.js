// eslint-disable-next-line import/newline-after-import
import Filters from './Filters.vue';
export default Filters;
