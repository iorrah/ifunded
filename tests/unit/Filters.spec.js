import { mount } from '@vue/test-utils';
import Filters from '../../src/components/Filters/index';

const props = {
  propsData: {
    appUsers: [
      {
        id: 1,
        first_name: 'Jane',
        last_name: 'Doe',
        email: 'jane.doeemail.com',
        created_date: new Date('2019-02-12'), // Feb
        creator_id: 1,
      },
      {
        id: 2,
        first_name: 'Bob',
        last_name: 'Aly',
        email: 'bob.aly@email.com',
        created_date: new Date('2019-12-02'), // Dec
        creator_id: 1,
      },
    ],
    creatorId: 1,
    month: 2,
    search: 'Doe',
  },
};

describe('Filters', () => {
  test('renders correctly', () => {
    const wrapper = mount(Filters, props);
    expect(wrapper.element).toMatchSnapshot();
  });
});
