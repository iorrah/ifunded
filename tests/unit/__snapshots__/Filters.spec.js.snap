// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters renders correctly 1`] = `
<div
  class="filters container"
>
  <p
    class="filters__title"
  >
    <img
      class="filters__icon"
      src="../../assets/filter-icon.svg"
    />
    
    Filter by
  
  </p>
   
  <div
    class="filters__inputs"
  >
    <div
      class="material-form"
    >
      <div
        class="material-form__input-text filters__input-text"
      >
        <div
          class="group"
        >
          <input
            class="inpField"
            data-message="The format of your email is incorrect"
            placeholder=""
            required="required"
            type="text"
          />
           
          <img
            class="material-form__search-icon"
            src="../../assets/search-icon.svg"
          />
           
          <span
            class="highlight"
          />
           
          <span
            class="bar"
          />
           
          <label>
            
            Name or email
          
          </label>
           
          <span
            class="error-message"
          >
            
            Error
          
          </span>
        </div>
      </div>
       
      <div
        class="material-form__input-select filters__input-select"
      >
        <div
          class="group select-group"
        >
          <div
            class="jq-selectbox jqselect inpField dropup opened"
          >
            <select
              class="inpField inpField__select"
              required="required"
            >
              <option
                label="Everyone"
                selected="selected"
                value="0"
              />
               
              <option
                value="1"
              >
                
                Jane Doe
              
              </option>
            </select>
             
            <div
              class="jq-selectbox__select"
            >
              <div
                class="jq-selectbox__select-text placeholder"
              >
                <span>
                  
                  Jane Doe
                
                </span>
              </div>
               
              <div
                class="jq-selectbox__trigger"
              >
                <div
                  class="jq-selectbox__trigger-arrow"
                >
                  <img
                    class="jq-selectbox__select-arrow"
                    src="../../assets/select-arrows.svg"
                  />
                </div>
              </div>
            </div>
          </div>
           
          <span
            class="highlight"
          />
           
          <span
            class="bar"
          />
        </div>
      </div>
       
      <div
        class="material-form__input-select filters__input-select"
      >
        <div
          class="group select-group"
        >
          <div
            class="jq-selectbox jqselect inpField dropup opened"
          >
            <select
              class="inpField inpField__select"
              required="required"
            >
              <option
                selected="selected"
                value="0"
              >
                Anytime
              </option>
               
              <option
                value="1"
              >
                January 2019
              </option>
              <option
                value="2"
              >
                February 2019
              </option>
              <option
                value="3"
              >
                March 2019
              </option>
              <option
                value="4"
              >
                April 2019
              </option>
              <option
                value="5"
              >
                May 2019
              </option>
              <option
                value="6"
              >
                June 2019
              </option>
              <option
                value="7"
              >
                July 2019
              </option>
              <option
                value="8"
              >
                August 2019
              </option>
              <option
                value="9"
              >
                September 2019
              </option>
              <option
                value="10"
              >
                October 2019
              </option>
              <option
                value="11"
              >
                November 2019
              </option>
              <option
                value="12"
              >
                December 2019
              </option>
            </select>
             
            <div
              class="jq-selectbox__select"
            >
              <div
                class="jq-selectbox__select-text placeholder"
              >
                <span>
                  
                  February 2019
                
                </span>
              </div>
               
              <div
                class="jq-selectbox__trigger"
              >
                <div
                  class="jq-selectbox__trigger-arrow"
                >
                  <img
                    class="jq-selectbox__select-arrow"
                    src="../../assets/select-arrows.svg"
                  />
                </div>
              </div>
            </div>
          </div>
           
          <span
            class="highlight"
          />
           
          <span
            class="bar"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
